prov/util/src/util_attr.c

```diff
@@ -40,15 +40,52 @@
 #define OFI_RMA_DIRECTION_CAPS	(FI_READ | FI_WRITE | \
 				 FI_REMOTE_READ | FI_REMOTE_WRITE)
 
+static inline bool is_sockaddr(uint32_t format)
+{
+	return format == FI_SOCKADDR || format == FI_SOCKADDR_IP ||
+	       format == FI_SOCKADDR_IN || format == FI_SOCKADDR_IN6 ||
+	       format == FI_SOCKADDR_IB;
+}
+
+static inline bool is_ipaddr(uint32_t format)
+{
+	return format == FI_SOCKADDR || format == FI_SOCKADDR_IP ||
+	       format == FI_SOCKADDR_IN || format == FI_SOCKADDR_IN6;
+}
+
+static inline bool is_ipv4addr(uint32_t format)
+{
+	return format == FI_SOCKADDR || format == FI_SOCKADDR_IP ||
+	       format == FI_SOCKADDR_IN;
+}
+
+static inline bool is_ipv6addr(uint32_t format)
+{
+	return format == FI_SOCKADDR || format == FI_SOCKADDR_IP ||
+	       format == FI_SOCKADDR_IN6;
+}
+
+static inline bool is_ibaddr(uint32_t format)
+{
+	return format == FI_SOCKADDR || format == FI_SOCKADDR_IB;
+}
+
+/*
+ * Used for filtering provider instances based on the address format. Expect
+ * exact match for formats that are not FI_SOCKADDR or FI_SOCKADDR_IP.
+ */
 int ofi_match_addr_format(uint32_t if_format, uint32_t user_format)
 {
 	if (user_format == FI_FORMAT_UNSPEC || if_format == FI_FORMAT_UNSPEC)
 		return 1;
 
 	switch (user_format) {
 	case FI_SOCKADDR:
-		/* Provider supports INET and INET6 */
-		return if_format <= FI_SOCKADDR_IN6;
+		/* matches INET, INET6, and IB */
+		return is_sockaddr(if_format);
+	case FI_SOCKADDR_IP:
+		/* matches INET and INET6 */
+		return is_ipaddr(if_format);
```

Comment:
nit: I'd drop the comments now that the semantics are consistent and you added the helper functions which make the checks clear.

Same comment on change below.